### PR TITLE
Move safe_decode_json to a Felix-independent module

### PR DESCRIPTION
**Pull request by neiljerram**
So that networking-calico can reliably use it

### CODE REVIEW

File: calico/etcdutils.py [hunk 1]
```diff
@@ -402,3 +405,77 @@ def delete_empty_parents(client, child_dir, root_key, timeout=DEFAULT_TIMEOUT):
...
+json_decoder = json.JSONDecoder(
+    object_hook=functools.partial(intern_dict,
+                                  fields_to_intern=FIELDS_TO_INTERN)
+)
+
+
+def safe_decode_json(raw_json, log_tag=None):
+    try:
+        return json_decoder.decode(raw_json)
+    except (TypeError, ValueError):
```
Comment (fasaxc) [inline calico/etcdutils.py]: I think you can get JSONDecodeError here too.  Not sure what causes that one but I think it's in the docs.

**Comment by fasaxc** [comment]
Are there any UTs of these functions?  If so, they're probably in the wrong file now.

File: calico/etcdutils.py (hunk 1 above)
Comment (neiljerram) [inline calico/etcdutils.py]: But I really don't think I should take that on as part of getting the new DHCP agent done.  Can you raise a Calico issue for this?  (Unless I've missed how this new point is related to my move...)

**Comment by neiljerram** [comment]
Sure, I'll look into that.

File: calico/etcdutils.py (hunk 1 above)
Comment (fasaxc) [inline calico/etcdutils.py]: OK

**Comment by neiljerram** [comment]
In fact there are no UTs of the functions that I've moved.  They are covered indirectly by higher-level Felix UTs.

**Comment by fasaxc** [comment]
Fine, don't think we need to add any just because they've been moved

**Comment by neiljerram** [comment]
Could this be merged now?

**Pull request merged**